### ISSUE TITLE
360safe browser needs promise polyfill 

### ISSUE THREAD

**Issue by jhiswin** [polyfillpolyfill/polyfill-service#500]
Uses a somewhat old version of a Chromium based engine.  Does not support native promises.
Has a very large amount of users in China.

**Comment by triblondon** [comment]
Hi @jhiswin, thanks for this.  The polyfill service uses the [useragent](https://github.com/3rd-Eden/useragent) module to identify browers, which ultimately uses the [browscap](https://github.com/browscap/browscap/tree/master/resources/user-agents/browsers) data.  This does not appear to include 360safe, so someone with knowledge of that browser will need to add it to browscap before we can support it in the polyfill service.

**Comment by mimmi20** [comment]
@triblondon, @jhiswin I understand the description of the [useragent](https://github.com/3rd-Eden/useragent) project, it uses data from [ua-parser](https://github.com/ua-parser/uap-core). Even if i'm wrong, its useful to open an issue there too.

**Comment by triblondon** [comment]
Ultimately it uses browscap. Start with an issue there.

**Comment by jhiswin** [comment]
@triblondon In your chart Chrome 31 had Promises, but testing on Browserstack I see that Chrome 31 did not have native Promises yet.  It was introduced in Chrome 32.  I think your polyfill needs to be adjusted?
So it looks like the polyfill service should actually work correctly with 360safe (which is detected as Chrome 31, which is what it actually is give or take a few features), it's just that the polyfill service has the Promise polyfill version requirement for Chrome incorrect by one version?

**Comment by triblondon** [comment]
Promise support in chrome sounds unrelated to 360safe, so please raise that
as a fresh issue. Happy to look into it.

**Comment by jhiswin** [comment]
@triblondon Yes, it looks as if this is an issue with Chrome 31.  So this is just a duplicate of the other issue, since 360safe is based on Chrome 31, and that is the reason for this issue.

**Comment by triblondon** [comment]
OK, if we idenfity 360safe as chrome, then that's fine.
